superduperdb/misc/superduper.py

```diff
@@ -3,13 +3,22 @@
 __all__ = ('superduper',)
 
 
-class DuckTyper:
+def superduper(item: t.Any, **kwargs) -> t.Any:
+    """
+    Attempts to automatically wrap an item in a superduperdb container using duck typing
+
+    :param item: A database or model
+    """
+    return _DuckTyper.run(item)
+
+
+class _DuckTyper:
     attrs: t.Sequence[str]
-    count: int = 0
+    count: int
 
-    @classmethod
-    def run(cls, item: t.Any, **kwargs) -> t.Any:
-        dts = [dt for dt in _DUCK_TYPES if dt.accept(item)]
+    @staticmethod
+    def run(item: t.Any, **kwargs) -> t.Any:
+        dts = [dt for dt in _DuckTyper._DUCK_TYPES if dt.accept(item)]
```

Comment:
This is perhaps outside the scope of this PR a little bit, but could we also have a more helpful error message for the case where `len(dts) > 1`?

Reply:
Done!